### PR TITLE
:sparkles: middleware helpers

### PR DESCRIPTION
**Pull request by guillaumecrespel**
#36

### CODE REVIEW

File: helpers/package.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "name": "trampss-redux-factory/helpers",
+  "private": true,
+  "main": "../helpers.js"
```
Comment (fabienjuif) [inline helpers/package.json]: why don't you build into `helpers` directory ?

File: src/helpers.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+export const mapAction = mapper => () => () => ctx => ({
+  ...ctx,
+  action: mapper(ctx.action),
+})
+
+export const mapPayload = actionMatches => mapper => () => () => (ctx) => {
+  const { payload, type } = ctx.action
+  if (!actionMatches || actionMatches.test(type)) {
+    return {
```
Comment (fabienjuif) [inline src/helpers.js]: `return mapAction(action => ({ ...action, payload: mapper(action.payload)()()(ctx)`

?

Comment (guillaumecrespel) [inline src/helpers.js]: :ok_hand: @fabienjuif

File: README.md [hunk 3]
```diff
@@ -176,5 +173,34 @@ import todos from './myTodosReducer'
 // state can be given by one of your middleware (redux-thunk, redux-saga, etc)
 // or it can be given by react-redux for example (mapStateToProps)
 todos.get('1')(state)
+```
+### helpers
+
+| signature | description | comment |
+|---|---|---|
+|`mapAction(<mapper(action)>)`| create middleware and map only redux action | function mapper(action) is mandatory |
```
Comment (delphinemillet) [inline README.md]: last column : `mapper(action)` is mandatory

Comment (guillaumecrespel) [inline README.md]: 👌 @delphinemillet

**Pull request merged**